### PR TITLE
Implemented: logic to create search preference and schedule a service (#29bdb9f)

### PR DESCRIPTION
**Pull request by ymaheshwari1**

### CODE REVIEW

File: src/components/SaveThresholdModal.vue [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      currentEComStore: 'user/getCurrentEComStore',
+      shopifyConfig: 'util/getShopifyConfig',
+      jobs: 'job/getJobs'
+    })
+  },
+  methods: {
+    closeModal() {
+      modalController.dismiss({ dismissed: true });
+    },
+    async createSearchPreference() {
```
Comment (adityasharma7) [inline src/components/SaveThresholdModal.vue:106]:
```suggestion
    async saveThresholdRule() {
```

File: src/store/modules/util/actions.ts [hunk 2]
```diff
@@ -28,6 +28,22 @@ const actions: ActionTree<UtilState, RootState> = {
...
   },
+
+  async getShopifyConfig({ commit }, payload) {
+    const resp = await UtilService.getShopifyConfig({
+      "inputFields": {
+        "productStoreId": payload
+      },
+      "entityName": "ShopifyConfig",
+      "noConditionFind": "Y",
+      "fieldList": ["shopifyConfigId", "productStoreId"]
```
Comment (adityasharma7) [inline src/store/modules/util/actions.ts:39]: Pass viewSize

File: src/views/SelectProduct.vue [hunk 3]
```diff
@@ -250,16 +234,58 @@ export default defineComponent({
...
           "query": "*:*",
-          "filter": "docType: PRODUCT"
+          "filter": ["docType: PRODUCT"]
         }
+      } as any
+    }
+  },
+  methods: {
+    updateInclusionQuery(value: string, type: string) {
+      this.included[type].includes(value) ? this.included[type].splice(this.included[type].indexOf(value), 1) : this.included[type].push(value)
```
Comment (adityasharma7) [inline src/views/SelectProduct.vue:244]: Assign this.included[type] to a variable and then use

File: src/views/SelectProduct.vue [hunk 4]
```diff
@@ -250,16 +234,58 @@ export default defineComponent({
...
+      this.excluded[type].includes(value) ? this.excluded[type].splice(this.excluded[type].indexOf(value), 1) : this.excluded[type].push(value)
+      this.updateQuery();
+    },
+    updateQuery() {
+      // initializing the filter always on updateQuery call because we are adding values in the filter
+      // as string and if some value is removed then we need to do multiple operations on the filter string
+      // to remove that value from the query filter
+      this.query.json['filter'] = ["docType: PRODUCT"]
+
+      this.included['tags'].length > 0 && this.query.json['filter'].push(`tags: (${this.included['tags'].join(' OR ')})`)
```
Comment (adityasharma7) [inline src/views/SelectProduct.vue:257]: We could iterate using reduce and prepare it dynamically

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/SelectProduct.vue (hunk 4 above)
Comment (ymaheshwari1) [reply src/views/SelectProduct.vue]: Sir, used reduce to prepare the query.

File: src/views/SelectProduct.vue (hunk 3 above)
Comment (ymaheshwari1) [reply src/views/SelectProduct.vue]: Done Sir

File: src/components/SaveThresholdModal.vue [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        <ion-label>56 {{ $t('SKUs') }}</ion-label>
+      </ion-item>
+      <ion-item>
+        <ion-icon :icon="optionsOutline" slot="start" />
+        <ion-label>{{ threshold ? threshold : '0' }} {{ $t('threshold') }}</ion-label>
+      </ion-item>
+
+      <ion-item>
+        <ion-label>{{ $t("Rule name") }}</ion-label>
+        <ion-input placeholder="rule name" v-model="jobName"/>
```
[the remark this answers is not in the document]
Comment (ymaheshwari1) [reply src/components/SaveThresholdModal.vue]: Added placeholder in internationalization file

File: src/components/SaveThresholdModal.vue (hunk 1 above)
Comment (ymaheshwari1) [reply src/components/SaveThresholdModal.vue]: Improved method name

File: src/store/modules/util/actions.ts (hunk 2 above)
Comment (ymaheshwari1) [reply src/store/modules/util/actions.ts]: For now added viewSize as 1.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**